### ISSUE TITLE
requests to JPEG files being massively flagged as vulnerable to csrf

### ISSUE THREAD

**Issue by starbuck3000** [ah8r/csrf#4]
Hi, 
I ran a scan this morning with the extension running and most JPEG files were flagged as vulnerable to csrf (rating: tentative). An option to disable this behavior or to setup some sort of threshold value (i.e.: don't report 'tentative' confidence level or lower) would definitely help.

**Comment by ah8r** [comment]
Would a feature similar to Burp's native "exclude from scope" be a solution here? So you could add a regex match for files ending in \.jpe?g and the extension would exclude them from scanning.

**Comment by starbuck3000** [comment]
I think that reusing the 'scope' semantics might confuse users, as we still want the extension to analyze all in-scope items.

The issue seems to be in the decision-logic on whether or not to flag a potential vulnerability. The  threshold is set at anything that does not contain variable values in a GET request, which (I think) is too sensitive. When using this extension, I get vulnerability reports for requests such as: 
- GET /
- GET /logo.png
- GET /styles.css

But not for a request such as:
- GET /impressum (<-- why isn't this flagged like the previous ones?)

I am still new using these extensions and I couldn't find a way in the documentation to create user-editable options for extensions, so I don't really see how you would do it differently than by changing the default threshold level?

**Comment by ah8r** [comment]
That's odd, because there's nothing in my current code which distinguishes /logo.png from /impressum so it should be reporting both, provided there are parameters in the request.

Can you post some example requests that get detected vs ones that aren't? If you can include the issue title as well (my extension reports several different issues) that would be helpful. This extension only works against requests with detected parameters (either URL or Body). The reasoning for this being that requests with no parameters can't be used to change state anyway. There are of course exceptions to this (/logout for example) but if the request is just to the URL http://example.com/image.jpg then it should not produce any issues. If however some form of anti-caching is being used and the URL is http://example.com/image.jpg?v=1.2 then that would be detected.

**Comment by starbuck3000** [comment]
Okay, I got it all wrong. I owe you an apology already, and an explanation of what happened. :)

I thought there were no "options" for extensions but I just noticed the 'csrf' tab for the first time (I am still convinced it wasn't there prior to the reboot but well). So I looked into it and carefully read the options available. Then I saw the "response forms" checkbox, which triggered my attention... Which led me into realizing that I had assumed half of your extensions' operation wrongly. I thought that the extension was reacting on whatever leaves the browser, but not that it could also flag solely based on the response.

That completely changes the situation: this is not a bug anymore but simply a feature request. I will open a separate ticket. My apologies again!
